src/product/tests/product.controller.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Test } from '@nestjs/testing';
+
+import { ProductController } from '../product.controller';
+import { ProductService } from '../product.service';
+import { CreateProductDto } from '../dto/create-product.dto';
+import { PaginationDto } from '../../common/dto/pagination.dto';
+import { Sorting } from '../../common/types/pagination';
+import { ProductState } from '../types/product';
+import { UpdateProductDto } from '../dto';
+
+describe('ProductController', () => {
+  let controller: ProductController;
+  let service: ProductService;
+
+  beforeEach(async () => {
+    const mockService = {
```

Comment:
WDYT about moving all product mocks to a separate file? It will keep the test cleaner and you will need in the future for other tests

Reply:
sure, it's done